crates/sui-indexer/src/store/indexer_store.rs

```diff
@@ -235,6 +235,10 @@ pub trait IndexerStore {
     ) -> Result<(), IndexerError>;
 
     async fn persist_epoch(&self, data: &TemporaryEpochStore) -> Result<(), IndexerError>;
+    async fn count_network_transaction_previous_epoch(
```

Comment:
I think a function that counts transactions in epoch X makes more sense than in epoch X-1

Reply:
this is to fetch the last `network_total_transactions` value of previous epoch in `checkpoints` table, we cannot count the tx of current epoch yet, b/c the checkpoint is likely not committed yet.

the more I think, the more I want to move all epoch related calculation to an async processor.

Reply:
I was suggesting:
```
fn count_network_transaction_previous_epoch
```

and then `call count_network_transaction_previous_epoch(n-1)`

will leave it up to you though

Reply:
yes, I saw what you meant, I wanted to calculate the total tx of current epoch based on
- committed row of last checkpoint of previous epoch, where the network_total_tx is tx count since genesis;
- the current checkpoint row in mem, which is the last row of the current epoch and may or may not committed

such that the difference is the tx count of current tx.

b/c the current checkpoint row might not have been committed, thus running query of `count_network_transaction_in_epoch`